### PR TITLE
Minor project cleanup 

### PR DESCRIPTION
**Pull request by lukasj**
* add missing serialVersionUID
* add missing comments
* turn build time warnings into errors
* add caching to the GH action

### CODE REVIEW

**Review by m0mus — APPROVED** [review]
LGTM

File: api/src/main/java/jakarta/json/JsonValueImpl.java [hunk 1]
```diff
@@ -26,8 +26,16 @@
...
 final class JsonValueImpl implements JsonValue, Serializable {
 
+    /** for serialization */
+    private static final long serialVersionUID = 83723433120886104L;
+
+    /** Type of this JsonValue. */
     private final ValueType valueType;
 
+    /**
+     * Default constructor.
```
Comment (pzygielo) [inline api/src/main/java/jakarta/json/JsonValueImpl.java:36]: I used to think that the default constructor is the one with no arguments...

Comment (lukasj) [reply api/src/main/java/jakarta/json/JsonValueImpl.java:36]: Why exactly the only one available cannot be the default one?

Comment (pzygielo) [reply api/src/main/java/jakarta/json/JsonValueImpl.java:36]: Yep, I was wrong: https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.8.9
So there is interpretation, that if defined - even no args one - it's no longer default (in Java).

**Pull request merged**